frontend/packages/metal3-plugin/src/components/host.tsx

```diff
@@ -118,13 +144,55 @@ type BaremetalHostsPageProps = {
   namespace: string;
 };
 
-export const BaremetalHostsPage = (props: BaremetalHostsPageProps) => (
-  <ListPage
-    {...props}
-    canCreate
-    rowFilters={filters}
-    createButtonText="Add Host"
-    kind={referenceForModel(BaremetalHostModel)}
-    ListComponent={HostList}
-  />
-);
+export const BaremetalHostsPage: React.FC<BaremetalHostsPageProps> = (props) => {
+  const hostsResource = {
+    kind: referenceForModel(BaremetalHostModel),
+    namespaced: true,
+    prop: 'hosts',
+  };
+  const machinesResource = {
+    kind: referenceForModel(MachineModel),
+    namespaced: true,
+    prop: 'machines',
+  };
+  const nodesResource = {
+    kind: NodeModel.kind,
+    namespaced: false,
+    prop: 'nodes',
+  };
+
+  const flatten = (resources) => {
+    // TODO(jtomasek): Remove loaded check once ListPageWrapper_ is updated to call flatten only
+    // when resources are loaded
```

Comment:
Are we sure we want to make that change in `ListPageWrapper_`? I'd think we'd want to show the data as it comes in. Otherwise we'd block rendering the table.

Reply:
I'll create a separate issue for this and try to elaborate on the issue I am seeing.